### PR TITLE
[Priest] Add Power Word: Shield insanity generation.

### PR DESCRIPTION
**Pull request by dj-riff**
This includes insanity generation from Hallucinations and Fae Blessings.

### CODE REVIEW

File: engine/class_modules/priest/sc_priest.cpp [hunk 1]
```diff
@@ -949,6 +974,8 @@ void priest_t::create_gains()
   gains.insanity_lost_devouring_plague         = get_gain( "Insanity spent on Devouring Plague" );
   gains.insanity_mindgames                     = get_gain( "Insanity Gained from Mindgames" );
   gains.insanity_ascended_blast                = get_gain( "Insanity Gained from Ascended Blast" );
+  gains.insanity_power_word_shield_hallucinations = get_gain( "Insanity Gained from Power Word: Shield" );
+  gains.insanity_night_fae					   = get_gain( "Insanity Gained from Power Word: Shield with Fae Blessings." );
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp]: this name should be more descriptive

File: engine/class_modules/priest/sc_priest.hpp [hunk 2]
```diff
@@ -319,6 +320,9 @@ struct priest_t final : public player_t
     propagate_const<gain_t*> insanity_lost_devouring_plague;
     propagate_const<gain_t*> insanity_mindgames;
     propagate_const<gain_t*> insanity_ascended_blast;
+	propagate_const<gain_t*> insanity_power_word_shield_hallucinations;
+	propagate_const<gain_t*> insanity_night_fae;
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.hpp]: fix indents

File: engine/class_modules/priest/sc_priest.cpp (hunk 1 above)
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp]: fix indents

File: engine/class_modules/priest/sc_priest.cpp [hunk 3]
```diff
@@ -661,37 +661,62 @@ struct shadow_mend_t final : public priest_heal_t
...
+			priest_absorb_t::execute();
+
+			if (priest().specialization() == PRIEST_SHADOW)
+			{
+				priest().generate_insanity(hallucinations_insanity_gain, priest().gains.insanity_power_word_shield_hallucinations, execute_state->action);
+			}
+
+		}
+		
+	};
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp]: this entire section is indented too much, and you are using too large of a gap with your tabbed lines (see above)

Also make sure when you use a paren you have proper spacing. `priest().genereate_insanity( x )`

**Review by seanpeters86 — CHANGES_REQUESTED** [review]

File: engine/class_modules/priest/sc_priest.cpp [hunk 4]
```diff
@@ -661,37 +661,62 @@ struct shadow_mend_t final : public priest_heal_t
...
+	// ==========================================================================
+	// Power Word: Shield
+	// ==========================================================================
+	struct power_word_shield_t final : public priest_absorb_t
+	{
+		double hallucinations_insanity_gain;
+		double nightfae_insanity_gain;
+		bool ignore_debuff;
+
+
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp]: extra line here

File: engine/class_modules/priest/sc_priest.hpp [hunk 5]
```diff
@@ -310,6 +311,10 @@ struct priest_t final : public player_t
     propagate_const<gain_t*> shadow_word_death_self_damage;
     propagate_const<gain_t*> insanity_death_and_madness;
     propagate_const<gain_t*> insanity_mindgames;
+    propagate_const<gain_t*> insanity_ascended_blast;
+	  propagate_const<gain_t*> insanity_power_word_shield_hallucinations;
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.hpp:315]: indent is wrong

File: engine/class_modules/priest/sc_priest.cpp [hunk 6]
```diff
@@ -927,7 +948,10 @@ void priest_t::create_gains()
   gains.insanity_death_and_madness             = get_gain( "Insanity Gained from Death and Madness" );
   gains.shadow_word_death_self_damage          = get_gain( "Shadow Word: Death self inflicted damage" );
   gains.insanity_mindgames                     = get_gain( "Insanity Gained from Mindgames" );
-  gains.insanity_eternal_call_to_the_void = get_gain( "Insanity Gained from Eternal Call to the Void Mind Flay's" );
+  gains.insanity_ascended_blast                = get_gain( "Insanity Gained from Ascended Blast" );
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp:951]: idk why this is added, probably a bad merge

File: engine/class_modules/priest/sc_priest.cpp [hunk 7]
```diff
@@ -652,37 +652,58 @@ struct shadow_mend_t final : public priest_heal_t
...
+      priest().buffs.fae_blessings->decrement();
+    }
   }
-};
+		
+  void execute() override
+  {
+	  priest_absorb_t::execute();
 
+    if (priest().specialization() == PRIEST_SHADOW)
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp:700]: this is wrong spacing 

```suggestion
    if ( priest().specialization() == PRIEST_SHADOW )
```

**Review by seanpeters86 — CHANGES_REQUESTED** [review]

**Comment by scamille** [comment]
The main problem is that you use tabs instead of spaces, see https://github.com/simulationcraft/simc/wiki/CodingGuidelines

But I would recommend that for the priest source code you just run clang-format on the files, and it does everything automatically for you. Most modern IDEs have a way of integrating clang-format, and then it is just one click to format the file you are currently working on.

**Comment by dj-riff** [comment]
I'm not sure what is happening with the tabs vs spaces. I manually spaced everything out so it conformed with the standards before I pushed anything.

> The main problem is that you use tabs instead of spaces, see https://github.com/simulationcraft/simc/wiki/CodingGuidelines
> 
> But I would recommend that for the priest source code you just run clang-format on the files, and it does everything automatically for you. Most modern IDEs have a way of integrating clang-format, and then it is just one click to format the file you are currently working on.

Clang was refusing to work in VS for me for some reason, which is the only reason I resorted to manually spacing it out.

**Comment by seanpeters86** [comment]
Fae Blessings has been changed, we no longer need this

**Pull request closed without merging**